compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/replacements/nodes/ArrayFillNode.java

```diff
@@ -0,0 +1,240 @@
+                    @ConstantNodeParameter JavaKind kind);
+
+    @NodeIntrinsic
+    public static native void fill(Pointer array, long offset, int length, int value,
+                    @ConstantNodeParameter JavaKind kind,
+                    @ConstantNodeParameter EnumSet<?> runtimeCheckedCPUFeatures);
+
+    @NodeIntrinsic
+    @GenerateStub(name = "longArrayFill", parameters = {"Long"})
+    @GenerateStub(name = "doubleArrayFill", parameters = {"Double"})
+    public static native void fill(Pointer array, long offset, int length, long value,
+                    @ConstantNodeParameter JavaKind kind);
+
+    @NodeIntrinsic
+    public static native void fill(Pointer array, long offset, int length, long value,
+                    @ConstantNodeParameter JavaKind kind,
+                    @ConstantNodeParameter EnumSet<?> runtimeCheckedCPUFeatures);
+
+    @Override
+    public ForeignCallDescriptor getForeignCallDescriptor() {
+        switch (this.elementKind) {
+            case Boolean:
+                return BOOLEAN_ARRAY_FILL_STUB;
+            case Byte:
+                return BYTE_ARRAY_FILL_STUB;
+            case Char:
+                return CHAR_ARRAY_FILL_STUB;
+            case Short:
+                return SHORT_ARRAY_FILL_STUB;
+            case Int:
+                return INT_ARRAY_FILL_STUB;
+            case Long:
+                return LONG_ARRAY_FILL_STUB;
+            case Float:
+                return FLOAT_ARRAY_FILL_STUB;
+            case Double:
+                return DOUBLE_ARRAY_FILL_STUB;
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public ValueNode[] getForeignCallArguments() {
+        return new ValueNode[]{arrayBase, offsetToFirstElement, arrayLength, valueToFillWith};
+    }
+
+    @Override
+    public void emitIntrinsic(NodeLIRBuilderTool gen) {
+        gen.getLIRGeneratorTool().emitArrayFill(elementKind, getRuntimeCheckedCPUFeatures(), gen.operand(arrayBase), gen.operand(offsetToFirstElement), gen.operand(arrayLength),
+                        gen.operand(valueToFillWith));
+    }
+
+    @Override
+    public LocationIdentity getLocationIdentity() {
+        return locationIdentity;
+    }
+
+    @Override
+    public boolean hasSideEffect() {
+        return true;
+    }
+
+    @Override
+    public LocationIdentity[] getKilledLocationIdentities() {
+        return new LocationIdentity[]{NamedLocationIdentity.getArrayLocation(this.elementKind)};
+    }
+
+    public static boolean isSupported(Architecture arch) {
+        return (arch instanceof AArch64);
+    }
+
+    @Override
+    public boolean canBeEmitted(Architecture arch) {
+        return isSupported(arch);
+    }
+
+    @Override
+    public FrameState stateAfter() {
+        return stateAfter;
+    }
+
+    @Override
+    public void setStateAfter(FrameState x) {
+        updateUsages(this.stateAfter, x);
+        this.stateAfter = x;
+    }
+
```

Comment:
Please add public getter methods `getArray()` and `getValue()` for the array object and the value to fill with.

Reply:
Actually, we will need getters for all four inputs plus the element kind.

Reply:
Done. Why do we need to add these?

Reply:
Thanks. The accessors are needed for internal changes in our Enterprise repositories for handling of this new node.